### PR TITLE
MINOR: Various cleanups in server and server-common

### PR DESCRIPTION
**Pull request by mimaison**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: server-common/src/test/java/org/apache/kafka/timeline/SnapshotRegistryTest.java [hunk 1]
```diff
@@ -41,9 +42,7 @@ public void testEmptyRegistry() {
     private static void assertIteratorContains(Iterator<Snapshot> iter,
                                                Snapshot... snapshots) {
         List<Snapshot> expected = new ArrayList<>();
```
Comment (chia7712) [inline server-common/src/test/java/org/apache/kafka/timeline/SnapshotRegistryTest.java]: How about `List<Snapshot> expected = Arrays.asList(snapshots);`?

File: server-common/src/test/java/org/apache/kafka/timeline/SnapshottableHashTableTest.java [hunk 2]
```diff
@@ -310,10 +309,8 @@ private static void assertIteratorYields(Iterator<?> iter,
             }
         }
         if (!extraObjects.isEmpty() || !remaining.isEmpty()) {
-            throw new RuntimeException("Found extra object(s): [" + String.join(", ",
-                extraObjects.stream().map(e -> e.toString()).collect(Collectors.toList())) +
-                "] and didn't find object(s): [" + String.join(", ",
-                remaining.keySet().stream().map(e -> e.toString()).collect(Collectors.toList())) + "]");
+            throw new RuntimeException("Found extra object(s): [" + extraObjects.stream().map(e -> e.toString()).collect(Collectors.joining(", ")) +
```
Comment (chia7712) [inline server-common/src/test/java/org/apache/kafka/timeline/SnapshottableHashTableTest.java]: `e -> e.toString()` -> `Object::toString`

File: server/src/main/java/org/apache/kafka/server/AssignmentsManager.java [hunk 3]
```diff
@@ -321,7 +321,7 @@ public void run() throws Exception {
                 AssignReplicasToDirsResponseData data = ((AssignReplicasToDirsResponse) response.responseBody()).data();
 
                 Set<AssignmentEvent> failed = filterFailures(data, inflight);
-                Set<AssignmentEvent> completed = Utils.diff(HashSet::new, inflight.values().stream().collect(Collectors.toSet()), failed);
+                Set<AssignmentEvent> completed = Utils.diff(HashSet::new, new HashSet<>(inflight.values()), failed);
```
Comment (chia7712) [inline server/src/main/java/org/apache/kafka/server/AssignmentsManager.java:324]: Maybe we don't need to create collection many times by skipping the failed elements.
```java
                Set<AssignmentEvent> failed = filterFailures(data, inflight);
                for (AssignmentEvent assignmentEvent : inflight.values()) {
                    if (failed.contains(assignmentEvent)) continue;
                    if (log.isDebugEnabled()) {
                        log.debug("Successfully propagated assignment {}", assignmentEvent);
                    }
                    assignmentEvent.onComplete();
                }
```

File: server/src/main/java/org/apache/kafka/server/metrics/ClientMetricsInstance.java [hunk 4]
```diff
@@ -116,8 +116,7 @@ public synchronized boolean maybeUpdatePushRequestTimestamp(long currentTime) {
         */
         boolean canAccept = lastGetRequestTimestamp > lastPushRequestTimestamp;
```
Comment (chia7712) [inline server/src/main/java/org/apache/kafka/server/metrics/ClientMetricsInstance.java:117]: how about `boolean canAccept = lastGetRequestTimestamp > lastPushRequestTimestamp || currentTime - lastPushRequestTimestamp >= pushIntervalMs;`?

**Review by chia7712 — COMMENTED** [review]
@mimaison nice cleanup. a couple of comments left

File: server/src/main/java/org/apache/kafka/server/AssignmentsManager.java (hunk 3 above)
Comment (mimaison) [reply server/src/main/java/org/apache/kafka/server/AssignmentsManager.java:324]: I'd prefer only sticking to simple refactorings in this PR. We can do this optimization in another PR.

File: server/src/main/java/org/apache/kafka/server/metrics/ClientMetricsInstance.java (hunk 4 above)
Comment (mimaison) [reply server/src/main/java/org/apache/kafka/server/metrics/ClientMetricsInstance.java:117]: It's shorter but I'm not sure it's more readable.

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request merged**